Kernel/Bus/USB/USBTransfer.h

```diff
@@ -35,19 +35,21 @@ class Transfer final : public AtomicRefCounted<Transfer> {
     USBRequestData const& request() const { return m_request; }
     Pipe const& pipe() const { return m_pipe; }
     Pipe& pipe() { return m_pipe; }
-    VirtualAddress buffer() const { return VirtualAddress(0xffff); }
-    PhysicalAddress buffer_physical() const { return PhysicalAddress(0xffff); }
+    VirtualAddress buffer() const { return m_dma_buffer->vaddr; }
+    PhysicalAddress buffer_physical() const { return m_dma_buffer->paddr; }
     u16 transfer_data_size() const { return m_transfer_data_size; }
     bool complete() const { return m_complete; }
     bool error_occurred() const { return m_error_occurred; }
 
 private:
-    Transfer(Pipe& pipe, u16 len);
+    Transfer(Pipe& pipe, u16 len, USBDMAPool<USBDMAHandle>& dma_pool, USBDMAHandle* dma_buffer);
     Pipe& m_pipe;                    // Pipe that initiated this transfer
     USBRequestData m_request;        // USB request
     u16 m_transfer_data_size { 0 };  // Size of the transfer's data stage
     bool m_complete { false };       // Has this transfer been completed?
     bool m_error_occurred { false }; // Did an error occur during this transfer?
+    USBDMAPool<USBDMAHandle>& m_dma_pool;
```

Comment:
The last commit is the one that actually makes USBTransfer functional with the new DmaPool and handles, is there a reason that the removal of the old member couldn't be deferred until the final commit, instead of breaking USBTransfer in the first commit and un-breaking it 3 commits later?

Reply:
The main reason was so I didn't have to carry around the current DMA buffer baggage in the form of parameters to different functions, only to remove it all after adapting it to fit the new Pipe code. I agree it's a little outside the scope of the first commit, but it would be a bit of a chore to go through and fix it up/test it with the old (current) buffer allocation code only to get rid of it again in the 4th commit. I'll let you make the judgement call on this one

Reply:
I went in and cleaned things up some more, so this should be resolved now and every commit should be atomic